server/apps/publish/archive_publish.py

```diff
@@ -245,8 +245,11 @@ def _publish_package_items(self, package, updates):
 
                     if package_item[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE:
                         # if the item is a package do recursion to publish
+                        wanted_keys = ['state', 'operation']
+                        sub_updates = dict([(i, updates[i]) for i in wanted_keys if i in updates])
```

Comment:
Creating a temporary list is not really needed, the same can be done more efficiently with a dict comprehension
```py
sub_updates = {i: updates[i] for i in wanted_keys if i in updates}
```